### PR TITLE
xmlto: 0.0.28 -> 0.0.29

### PR DESCRIPTION
**Pull request by trofi**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/typesetting/xmlto/default.nix [hunk 1]
```diff
@@ -64,6 +52,7 @@ stdenv.mkDerivation (finalAttrs: {
   # `libxml2' provides `xmllint', needed at build-time and run-time.
   # `libxslt' provides `xsltproc', used by `xmlto' at run-time.
   nativeBuildInputs = [
+    autoreconfHook
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/xmlto/default.nix:55]: infinite recursion...

Comment (trofi) [reply pkgs/tools/typesetting/xmlto/default.nix:55]: I wonder what system it fails the eval on and what attribute it is. All I try seems to work when merged to current `staging`:

```
$ for system in {x86_64,aarch64}-{linux,darwin}; do nix-instantiate -A xmlto --argstr system $system; done

/nix/store/gj6j9jycnvv2iz9n0byxmci36h4h7i8m-xmlto-0.0.29.drv
/nix/store/6263sr6ngy3gmqgqngjaksijjcdkyix0-xmlto-0.0.29.drv
/nix/store/645lhdwl34dnswlbnz12nfx71ir9ppb0-xmlto-0.0.29.drv
/nix/store/6xzlr847gmiw8kq54njdwkws40dwb0rb-xmlto-0.0.29.drv
```

Comment (SuperSandro2000) [reply pkgs/tools/typesetting/xmlto/default.nix:55]: It fails through git. see the full ofborg log for the details

Comment (trofi) [reply pkgs/tools/typesetting/xmlto/default.nix:55]: Does it fail for you as well? What should I run locally to see the same failure?

Comment (SuperSandro2000) [reply pkgs/tools/typesetting/xmlto/default.nix:55]: I didn't try it or locally but I would expect building git or gitFull would fail.

Comment (trofi) [reply pkgs/tools/typesetting/xmlto/default.nix:55]: I would expect `xmlto` itself to fail as well in that case. Tried to instantiate `git` and `gitFull` when merged this `PR` against `staging`. Instantiates just fine:

```
$ for system in {x86_64,aarch64}-{linux,darwin}; do nix-instantiate -A git -A gitFull --argstr system $system; done

/nix/store/nsbwv0n1hqjfmlfcivr7yrhimghxr8fm-git-2.45.2.drv
/nix/store/dybj21wa71zmv9iqn2v6zqc3xnb060cr-git-with-svn-2.45.2.drv

/nix/store/g0g8vpprs03ykd9jjdma0s575xjkdrvs-git-2.45.2.drv
/nix/store/wazdz8knx07ggyy56860xf9i3pb3k56v-git-with-svn-2.45.2.drv

/nix/store/qhjzf8qdcfshsiw6pz0r1blg34k385s5-git-2.45.2.drv
/nix/store/0dqaf7yhvqm1b57jhn4nrg21k89b59r4-git-with-svn-2.45.2.drv

/nix/store/xw734k1v52dhac1mk67psimzw415rhdp-git-2.45.2.drv
/nix/store/0ih7aa39fmw4szicr46g103l4li83a02-git-with-svn-2.45.2.drv
```

I suspect the eval failure was done against some stale state of `staging`.

**Comment by trofi** [comment]
Rebased against fresher merge-base between `staging` and `master` to poke `ofborg` for re-eval.

File: pkgs/tools/typesetting/xmlto/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/typesetting/xmlto/default.nix:54]: > I suspect the eval failure was done against some stale state of staging.

That seems to have been the root cause.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**